examples/droplet/main.tf

```
@@ -6,39 +15,40 @@ provider "digitalocean" {
 
 resource "digitalocean_droplet" "mywebserver" {
   # Obtain your ssh_key id number via your account. See Document https://developers.digitalocean.com/documentation/v2/#list-all-keys
-  ssh_keys           = [12345678]         # Key example
-  image              = "${var.ubuntu}"
-  region             = "${var.do_ams3}"
+  ssh_keys           = [29834845]        # Key example
+  image              = var.ubuntu
+  region             = var.do_ams3
   size               = "s-1vcpu-1gb"
   private_networking = true
   backups            = true
   ipv6               = true
   name               = "mywebserver-ams3"
+  user_data          = file("cloudinit.conf")
 
-  provisioner "remote-exec" {
-    inline = [
-      "export PATH=$PATH:/usr/bin",
-      "sudo apt-get update",
-      "sudo apt-get -y install nginx",
-    ]
-
-    connection {
+  connection {
+      host     = self.ipv4_address
       type     = "ssh"
-      private_key = "${file("~/.ssh/id_rsa")}"
+      private_key = file("~/.ssh/id_rsa")
       user     = "root"
       timeout  = "2m"
     }
-  }
 }
 
```

Adding an ssh key resource allows the user to upload their real ssh key based on the input variable.
```suggestion

resource "digitalocean_ssh_key" "example" {
  name       = "examplekey"
  public_key = file(var.ssh_key_path)
}

```